api/status.go

```diff
@@ -0,0 +1,209 @@
+			case types.LXD:
+				lxd, err = s.(*service.LXDService).Client(context.Background())
+			case types.MicroCeph:
+				microClient, err = s.(*service.CephService).Client("")
+			case types.MicroOVN:
+				microClient, err = s.(*service.OVNService).Client()
+			case types.MicroCloud:
+				microClient, err = s.(*service.CloudService).Client()
+			}
+			if err != nil {
+				return err
+			}
+
+			if microClient != nil {
+				clusterMembers, err := microClient.GetClusterMembers(context.Background())
+				if err != nil && !lxdAPI.StatusErrorCheck(err, http.StatusServiceUnavailable) {
+					return err
+				}
+
+				status.Clusters[s.Type()] = clusterMembers
+
+				if s.Type() == types.MicroCeph {
+					disks, err := cephClient.GetDisks(r.Context(), microClient)
+					if err != nil {
+						return err
+					}
+
+					for _, disk := range disks {
+						if disk.Location == s.Name() {
+							status.OSDs = append(status.OSDs, disk)
+						}
+					}
+
+					services, err := cephClient.GetServices(r.Context(), microClient)
+					if err != nil {
+						return err
+					}
+
+					for _, service := range services {
+						if service.Location == s.Name() {
+							status.CephServices = append(status.CephServices, service)
+						}
+					}
+				}
+
+				if s.Type() == types.MicroOVN {
+					services, err := ovnClient.GetServices(r.Context(), microClient)
+					if err != nil {
+						return err
+					}
+
+					for _, service := range services {
+						if service.Location == s.Name() {
+							status.OVNServices = append(status.OVNServices, service)
+						}
+					}
+				}
+
+			} else if lxd != nil {
+				server, _, err := lxd.GetServer()
+				if err != nil {
+					return err
+				}
+
+				if server.Environment.ServerClustered {
+					clusterMembers, err := lxd.GetClusterMembers()
+					if err != nil {
+						return err
+					}
+
+					certs, err := lxd.GetCertificates()
+					if err != nil {
+						return err
+					}
+
+					microMembers := make([]microTypes.ClusterMember, 0, len(clusterMembers))
+					for _, member := range clusterMembers {
+						url, err := url.Parse(member.URL)
+						if err != nil {
+							return err
+						}
+
+						addrPort, err := microTypes.ParseAddrPort(util.CanonicalNetworkAddress(url.Host, 8443))
+						if err != nil {
+							return err
+						}
+
+						// Microcluster requires a certificate to be specified in types.ClusterMemberLocal.
+						var serverCert *microTypes.X509Certificate
+						for _, cert := range certs {
+							if cert.Type == "server" && cert.Name == member.ServerName {
+								serverCert, err = microTypes.ParseX509Certificate(cert.Certificate)
+								if err != nil {
+									return err
+								}
+							}
+						}
+
+						microMember := microTypes.ClusterMember{
+							ClusterMemberLocal: microTypes.ClusterMemberLocal{
+								Name:        member.ServerName,
+								Address:     addrPort,
+								Certificate: *serverCert,
+							},
+							Role:       strings.Join(member.Roles, ","),
```

Comment:
Don't we display those roles in the final table as these are the internal Raft ones and we don't have a concept like `mds` or `central` in LXD like for the other Micro* snaps?

Reply:
Same here, the role is supplied for consistency because it's automatically supplied for the `microcluster` projects.

Reply:
I see thanks.